app/controllers/podcast_metrics_controller.rb

```diff
@@ -3,13 +3,31 @@ class PodcastMetricsController < ApplicationController
 
   before_action :set_podcast
   before_action :check_clickhouse, except: %i[show]
-  before_action :set_date_range, except: %i[dropdays]
-  before_action :set_uniques, only: %i[show uniques]
-  before_action :set_dropday_range, only: %i[show dropdays]
 
   def show
   end
 
+  def episode_sparkline
+    @episode = Episode.find_by(guid: metrics_params[:episode_id])
+    @prev_episode = Episode.find_by(guid: metrics_params[:prev_episode_id])
+
+    @episode_trend = calculate_episode_trend(@episode, @prev_episode)
+    @sparkline_downloads =
+      Rollups::HourlyDownload
+        .where(episode_id: @episode[:guid], hour: (@episode.first_rss_published_at..))
```

Comment:
I think you need to set the lowerbound to the start of the hour... otherwise you'll miss out on the first hour.

Imagine we wrote the RSS at 5 minutes after the hour - there could be a substantial spike for those next 55 minutes.

Reply:
Also, can we cap this at 6 months of data or something?  I happened to pull up Transistor (last episode published in 2017)... and this was a lot of data!